pkgs/applications/science/physics/freefem/default.nix

```diff
@@ -0,0 +1,82 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, autoreconfHook
+, bison
+, flex
+, unzip
+, gfortran
+, lapack
+, hdf5-cpp
+, scotch
+, fftw
+, arpack
+, suitesparse
+, nlopt
+, ipopt
+, gmm
+, tetgen
+, gnuplot_qt
+}:
+stdenv.mkDerivation rec {
+  pname = "freefem";
+  version = "4.12";
+
+  src = fetchFromGitHub {
+    owner = "FreeFem";
+    repo = "FreeFem-sources";
+    rev = "v${version}";
+    sha256 = "sha256-6EdkXaWf76wODfKwKw2gKRLY9hpcHUzAgtTSdQeHSoU=";
+  };
+
+  nativeBuildInputs = [
+    autoreconfHook
+    bison
+    flex
+    unzip
+    gfortran
+  ];
+
+  buildInputs = [
+    lapack
+    hdf5-cpp
+    scotch
+    fftw
+    arpack
+    suitesparse
+    nlopt
+    ipopt
+    gmm
+    tetgen
+    gnuplot_qt
+  ];
+
+  outputs = [ "out" ];
+  enableParallelBuilding = true;
+
+  configureFlags = [ "--enable-optim" ];
+
+  # Currently this fails with 1 test
+  # doCheck = true;
+
+  postPatch = ''
+    patchShebangs .
+  '';
+
+  meta = with lib; {
+    description = "A high level multiphysics finite element software";
+    longDescription = ''
+      FreeFEM is a popular 2D and 3D partial differential equations (PDE)
+      solver used by thousands of researchers across the world.
+
+      It allows you to easily implement your own physics modules using the
+      provided FreeFEM language. FreeFEM offers a large list of finite
+      elements, like the Lagrange, Taylor-Hood, etc., usable in the
+      continuous and discontinuous Galerkin method framework.
+    '';
+    homepage = "https://freefem.org";
+    changelog = "https://github.com/FreeFem/FreeFem-sources/blob/master/CHANGELOG.md";
+    platforms = platforms.linux;
+    license = licenses.gpl3;
```

Comment:
It appears to be under `lgpl3Plus`, checked https://github.com/FreeFem/FreeFem-sources/blob/master/LICENSE.md and https://src.fedoraproject.org/rpms/freefem++/blob/rawhide/f/freefem++.spec